Automattic/jetpack
# [Boost] Migrate Speed Score popups to React

Port the Speed Score popout to React.

A few notes:
- Adds `react-transition-group` to handle transitioning the popout in / out from the side.
- Alters the popout logic to always render the last message shown so that the popout can transition out in a react-friendly manner
- Switches from using a link for the dismiss option, to using a button for a11y compliance.

## Proposed changes:
* Port the Speed Score popout to React

### Other information:

## Jetpack product discussion


pc9hqz-2jy-p2

## Does this pull request change what data or activity we track or use?


no

## Testing instructions:





* Make sure you clear any previously popup dismissals, to make sure this appears; `delete from wp_options where option_name='jetpack_boost_ds_dismissed_score_prompt' or option_name='jb_show_score_prompt';`
* Force your speed score to show a large increase, by adding this to `Speed_Score::fetch_speed_score`:
```
return [
	'status' => 'success',
	'scores' => [
		'current' => [
			'desktop' => 100,
			'mobile' => 100,
		],
		'noBoost' => [
			'desktop' => 50,
			'mobile' => 50,
		],
		'isStale' => false,
	]
];
```
* Load the Boost dashboard, make sure the score increase popup appears
* Close the popup with the X button and refresh the score, make sure the popup appears again
* Make sure that clicking either the rate button or the dismiss link makes the dialog go away (and not return even on refresh)
* Do the same for score decrease
